### PR TITLE
Add new "find" filter

### PR DESCRIPTION
**Pull request by smnandre**
This filter returns the first _value_ matching the given arrow function or null

Attempt to solve #3962 

(open to any feedback)

### CODE REVIEW

**Comment by smnandre** [comment]
(failures unrelated with this filter)

**Comment by smnandre** [comment]
Added documentation

(fabbot still complaining on a false-positive)

File: doc/filters/find.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+``find``
+==========
```
Comment (fabpot) [inline doc/filters/find.rst]:
```suggestion
========
```

File: doc/filters/find.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+The ``find`` returns the first element of a sequence matching an arrow
```
Comment (fabpot) [inline doc/filters/find.rst]:
```suggestion
The ``find`` filter returns the first element of a sequence matching an arrow
```

File: doc/filters/find.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+The arrow function also receives the key as a second argument:
+
+.. code-block:: twig
+
+    {{ sizes|find((v, k) => 's' not in k) }}
+
+    {# output 38 #}
+
+Note that the arrow function has access to the current context.
```
Comment (fabpot) [inline doc/filters/find.rst]: Maybe we can give an example on how to do it.

File: doc/filters/find.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
+``find``
+==========
+
```
Comment (fabpot) [inline doc/filters/find.rst:3]: You should add the following:

```rst
.. versionadded:: 3.11

    The ``find`` function was added in Twig 3.11.
```

File: src/Extension/CoreExtension.php [hunk 5]
```diff
@@ -218,6 +218,7 @@ public function getFilters(): array
             new TwigFilter('filter', [self::class, 'filter'], ['needs_environment' => true]),
             new TwigFilter('map', [self::class, 'map'], ['needs_environment' => true]),
             new TwigFilter('reduce', [self::class, 'reduce'], ['needs_environment' => true]),
+            new TwigFilter('find', [self::class, 'arrayFind'], ['needs_environment' => true]),
```
Comment (fabpot) [inline src/Extension/CoreExtension.php]:
```suggestion
            new TwigFilter('find', [self::class, 'find'], ['needs_environment' => true]),
```

**Review by fabpot — CHANGES_REQUESTED** [review]
Can you also update the CHANGELOG file?

**Comment by smnandre** [comment]
> Can you also update the CHANGELOG file?

Good. Any other need in documentation or tests ?

**Comment by fabpot** [comment]
Thank you @smnandre.

**Pull request merged**